### ISSUE TITLE
Stick ability

### ISSUE THREAD

**Issue by adriaaula** [KucherenkoSerhiy/Microworld#5]
This is enemy specific, since the counterpart (when the P2 controlls the enemy) will be the WallJump ability, one of the special abilities of the specific enemy type. 

It is already implemented by Sergio, it is just necessary to convert it to the new structure.

**Comment by scastlara** [comment]
I understand that the "Stick Ability" only includes the "_I will stick to you and impede your movement_" of it, and not the "_I will transform myself into an enemy spawner_".

**Comment by AdriaPerezCulubret** [comment]
Yes, you understand correctly.

**Comment by adriaaula** [comment]
Yes, although If we consider that both abilities are the ones from that specific enemy, they could come together. 

The thing is that when we talked about this, we said that we could define one ability, used both by the P2 (through controllers when possesed) and from the enemy. 

But since the enemy will have quite a different behavior (since the ability will be designed to be used when a player is playing), maybe we can have an specific Ability useful for the enemy, in which it works slightly similar// slightly different than the one from the P2. 

For example this: Walljump for P2, since it is able to stick to the walls through one key, and StickEnemy, only able to stick to the players or other creatures, and reproducing by hurting the creatures.

**Comment by KucherenkoSerhiy** [comment]
Abilities are designed to be used independentely of environment. For deciding WHEN, use control class that can represent as player as bot

**Comment by adriaaula** [comment]
Ok, this is a topic we should discuss next time we meet.

**Comment by scastlara** [comment]
New branch "feature/stick-ability".

As of now, _StickAbility_ changes the _MoveSpeed_ of _Stickable_ objects upon collision, by dividing it by _StickAbilityArgs.SlowingFactor_. A boolean _StickAbilityArgs.IsSticked_ is set to true, to avoid sticking to more than one object at a time.

The "only" thing missing is that the Object with the StickAbility has to change its behaviour in some way, to assure that it remains "sticked" to the colliding object. Doing this is not trivial, and I might need some guidance.

Cheers.
